File: Part3/test/T3L100_ok.pas
(* Test file for 100% level of CSCE 531 Pascal compiler Project III *)

(* There are no errors in this file *)

program T3L100_ok;

var
   i,j,k : Integer;
   f	 : Single;
   c	 : Char;
   iaa	 : array[0..13, 0..11] of Integer;
   raaa	 : array[2..4, 0..5, 1..4] of Real;
   r	 : Real;


procedure print_nl; External;
procedure print_dot; External;
procedure print_even; External;
procedure print_odd; External;
procedure print_j; External;
procedure print_i_j_values; External;
procedure print_oops; External;
procedure print_iaa; External;
procedure print_raaa; External;
procedure print_r; External;


begin
   for i := 1 to 10 do
      print_dot;
   print_nl;

   i := 15;
   for i := 1 to i do
      print_dot;
   print_nl;

   for c := chr(1) to chr(10) do
      print_dot;
   print_nl;

   for i := 0 to 9 do begin
      case i of
	0,2,4,6,8: print_even;
	1,3,5,7,9: print_odd
      end;
      if i = 1 then i := 3;
      for j := 1 to i-1 do
	 print_j;
   end;
   print_nl;
   print_i_j_values;

   for j := 10 downto 1 do
      print_i_j_values;

   for i := 0 to 9 do;
   print_oops;

   (* Pascal's triangle again, this time with a 2-d array *)
   for j := 0 to 11 do
      iaa[0,j] := 1;
   for i := 1 to 13 do begin
      iaa[i,0] := 1;
      for j := 1 to 11 do
	 iaa[i,j] := (iaa[i,j-1] + iaa[i-1,j]) mod 10000
   end;

   print_iaa;

   r := 9.0;
   for i := 2 to 4 do
      for j := 0 to 5 do
	 for k := 1 to 4 do begin
	    raaa[i,j,k] := r;
	    r := r * 0.9
	 end;

   print_raaa;

   i := 1;
   j := 2;
   r := raaa[i*3,2*j,3];

   print_r;

end.

File: Part3/T3L100_ok.c
long int I,J,K;
float F;
unsigned char C;
long int Iaa[14][12];  // Indices start at 0, 0
double Raaa[3][6][4];  // indices start at 2, 0, 1
double R;

void Print_nl();
void Print_dot();
void Print_even();
void Print_odd();
void Print_j();
void Print_i_j_values();
void Print_oops();
void Print_iaa();
void Print_raaa();
void Print_r();


int main()
{
    for (I = 1; I <= 10; I++)
        Print_dot();
    Print_nl();

    I = 15;
    for (I = 1; I <= 15; I++)
        Print_dot();
    Print_nl();

    for (C = 1; C <= 10; C++)
        Print_dot();
    Print_nl();

    for (I = 0; I <= 9; I++) {
        switch (I) {
            case 0:
            case 2:
            case 4:
            case 6:
            case 8:
                Print_even(); break;
            case 1:
            case 3:
            case 5:
            case 7:
            case 9:
                Print_odd();
        }
        if (I == 1) I = 3;
        for (J = 1; J <= I-1; J++)
            Print_j();
    }
    Print_nl();
    Print_i_j_values();

    for (J = 10; J >= 1; J--)
        Print_i_j_values();

    for (I = 0; I <= 9; I++);
    Print_oops();

        /* Pascal's triangle again, this time with a 2-d array */
    for (J = 0; J <= 11; J++)
        Iaa[0][J] = 1;
    for (I = 1; I <= 13; I++) {
        Iaa[I][0] = 1;
        for (J = 1; J <= 11; J++)
            Iaa[I][J] = (Iaa[I][J-1] + Iaa[I-1][J]) % 10000;
    }

    Print_iaa();

    R = 9.0;
    for (I = 2; I <= 4; I++)
        for (J = 0; J <= 5; J++)
            for (K = 1; K <= 4; K++) {
                Raaa[I-2][J][K-1] = R;
                R = R * 0.9;
            }

    Print_raaa();

    I = 1;
    J = 2;
    R = Raaa[I*3-2][2*J][3-1];

    Print_r();
    return 0;
}
